src/mergecil.ml

```diff
@@ -1382,7 +1384,7 @@
        | GVar (vi, init, l) ->
            currentLoc := l;
            incr currentDeclIdx;
-           let vi' = processVarinfo vi l in
+           let vi' = processVarinfo ~isadef:(init.init <> None) vi l in
```

Comment:
This is just matching the login in Cabs2cil:
https://github.com/goblint/cil/blob/986a509d85417a55b4c98e635e3d52abb7e85b3f/src/frontc/cabs2cil.ml#L5642
https://github.com/goblint/cil/blob/986a509d85417a55b4c98e635e3d52abb7e85b3f/src/frontc/cabs2cil.ml#L1861

Also, there's some Mergecil logic that keeps `GVar`s without initializers around instead of `GVarDecl`s:
https://github.com/goblint/cil/blob/986a509d85417a55b4c98e635e3d52abb7e85b3f/src/mergecil.ml#L1399

Reply:
Ok, seems reasonable then.